tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java

```diff
@@ -0,0 +1,439 @@
+
+        final SubscribeStreamResponse subscribeResponse1 =
+                subscribeResponseObserver.getOnNextCalls().get(2);
+        assertThat(subscribeResponse1.blockItems().blockItems()).hasSize(blockItems1.length);
+
+        assertThat(subscribeResponseObserver.getOnNextCalls())
+                .hasSize(3) // block 0 items, success status and block 1 items
+                .element(2)
+                .satisfies(response -> {
+                    assertThat(response.blockItems().blockItems()).hasSize(blockItems1.length);
+                    assertThat(response.blockItems()
+                                    .blockItems()
+                                    .getFirst()
+                                    .blockHeader()
+                                    .number())
+                            .isEqualTo(blockNumber1);
+                });
+
+        // close the client connections
+        blockStreamPublishServiceClient.close();
+        blockStreamSubscribeServiceClient.close();
+        blockAccessServiceClient.close();
+        blockNodeServiceClient.close();
+    }
+
+    @Disabled
+    @Test
+    void swappedPublisherConnection() {
+        BlockStreamPublishServiceInterface.BlockStreamPublishServiceClient initialPublishClient =
+                new BlockStreamPublishServiceInterface.BlockStreamPublishServiceClient(
+                        publishBlockStreamPbjGrpcClient, OPTIONS);
+
+        ResponsePipelineUtils<PublishStreamResponse> initialResponseObserver = new ResponsePipelineUtils<>();
+        final Pipeline<? super PublishStreamRequest> requestStream =
+                initialPublishClient.publishBlockStream(initialResponseObserver);
+
+        final long blockNumber = 0;
+        BlockItem[] blockItems = BlockItemBuilderUtils.createSimpleBlockWithNumber(blockNumber);
+        // build request with incomplete block (missing proof)
+        PublishStreamRequest request = PublishStreamRequest.newBuilder()
+                .blockItems(BlockItemSet.newBuilder()
+                        .blockItems(blockItems[0], blockItems[1])
+                        .build())
+                .build();
+
+        requestStream.onNext(request);
+
+        // sleep briefly to allow processing
+        parkNanos(5_000_000_000L);
+
+        assertThat(initialResponseObserver.getOnNextCalls()).hasSize(0); // no responses should be sent yet
+
+        // Assert no other responses sent
+        assertThat(initialResponseObserver.getOnErrorCalls()).isEmpty();
+        assertThat(initialResponseObserver.getOnSubscriptionCalls()).isEmpty();
+        assertThat(initialResponseObserver.getOnCompleteCalls().get()).isEqualTo(0);
+        assertThat(initialResponseObserver.getClientEndStreamCalls().get()).isEqualTo(0);
+
+        // close the initial connection
+        publishBlockStreamPbjGrpcClient.close(); // to-do: figure out why this close is not triggering a block close
```

Comment:
after rebase, if you enable this test and add this:


```suggestion
        // send endStream before closing connection
        PublishStreamRequest endStreamRequest = PublishStreamRequest.newBuilder()
                .endStream(PublishStreamRequest.EndStream.newBuilder().endCode(PublishStreamRequest.EndStream.Code.RESET)
                        .build())
                .build();
        requestStream.onNext(endStreamRequest);

        publishBlockStreamPbjGrpcClient.close(); // to-do: figure out why this close is not triggering a block close
```

and the next comment:

Reply:
Note, this requires the work _planned_ under Issue #1239, it won't work _yet_.